packages/clerk-js/src/core/clerk.ts

```diff
@@ -1407,13 +1419,24 @@ export class Clerk implements ClerkInterface {
       const tracker = createBeforeUnloadTracker(this.#options.standardBrowser);
 
       await tracker.track(async () => {
-        await this.navigate(redirectUrlComplete);
+        if (!this.client) {
+          return;
+        }
+
+        if (this.client.isEligibleForTouch()) {
+          const absoluteRedirectUrl = new URL(redirectUrlComplete, window.location.href);
+          await this.navigate(this.buildUrlWithAuth(this.client.buildTouchUrl({ redirectUrl: absoluteRedirectUrl })));
+        } else {
+          await this.navigate(redirectUrlComplete);
+        }
```

Comment:
We weren't dealing with this case before, even tho I never caught it, but it's the same handling as `setActive({ redirectUrl: this.afterSignInUrl() }` used to do. 

If the client cookie is about to expire upon redirection, then we must decorate the URL with auth parameters.

Reply:
This also has started to break some unit tests since `this.buildUrlWithAuth` is not being mocked, I'll fix it in a bit.